hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -260,15 +260,22 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIn
               " metadata while recovery flag is not set in request", KEY_UNDER_LEASE_RECOVERY);
         }
       }
-      omKeyInfo.getMetadata().putAll(KeyValueUtil.getFromProtobuf(
-          commitKeyArgs.getMetadataList()));
+
+      // non-null indicates it is necessary to update the open key
+      OmKeyInfo newOpenKeyInfo = null;
 
       if (isHSync) {
-        omKeyInfo.getMetadata().put(OzoneConsts.HSYNC_CLIENT_ID, clientIdString);
+        if (!OmKeyHSyncUtil.isHSyncedPreviously(omKeyInfo, clientIdString, dbOpenKey)) {
+          omKeyInfo.getMetadata().put(OzoneConsts.HSYNC_CLIENT_ID, clientIdString);
+          newOpenKeyInfo = omKeyInfo.copyObject();
+        }
       } else if (isRecovery) {
         omKeyInfo.getMetadata().remove(OzoneConsts.HSYNC_CLIENT_ID);
         omKeyInfo.getMetadata().remove(OzoneConsts.LEASE_RECOVERY);
       }
+
+      omKeyInfo.getMetadata().putAll(KeyValueUtil.getFromProtobuf(
+          commitKeyArgs.getMetadataList()));
```

Comment:
Please move this metadata.putAll before if(isHsync),  I believe the commitKeyArgs will have HSYNC_CLIENT_ID and 
LEASE_RECOVERY metadata, so if you put it here, then the final committed key will still have these two metadata.

Reply:
Or you can remove L272-275, and add remove both HSYNC_CLIENT_ID and LEASE_RECOVERY in L349.

Reply:
Done. Also kept the original semantic for `LEASE_RECOVERY` by keeping `if (isRecovery)` check.